corehq/apps/app_manager/views/utils.py

```diff
@@ -172,17 +172,21 @@ def overwrite_app(app, master_build, report_map=None):
     # so the linked app needs to know if there are any new forms and, if so, assign those forms new unique ids.
     # To do this lookup, get the XMLNSes from the the most recent versions of this app pulled from each master
     # and compare those to the XMLNSes present in this app.
+    unknown_xmlnses = {form.xmlns for form in app.get_forms()}
     form_ids_by_xmlns = {}
-    master_app_briefs = app.get_master_app_briefs()
-    for brief in master_app_briefs:
-        previous_app = app.get_latest_build_from_upstream(brief.master_id)
-        if previous_app:
-            form_ids_by_xmlns.update(_get_form_ids_by_xmlns(previous_app))
+    unknown_xmlnses = unknown_xmlnses.difference(form_ids_by_xmlns.keys())
```

Comment:
nit: this line isn't doing anything since `form_ids_by_xmlns` is an empty dict here

Reply:
e5ac4ba227d